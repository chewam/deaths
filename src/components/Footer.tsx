import Lang from "@/components/Lang"
import Switch from "@/components/Switch"
import { FormattedMessage as Trans } from "react-intl"

const Spacer = (): JSX.Element => (
  <span className="spacer">&nbsp;&nbsp;|&nbsp;&nbsp;</span>
)

const Footer = (): JSX.Element => {
  const version = process.env.APP_VERSION || "0.0.0"
  const sha = process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA || "dev"

  return (
    <footer>
      <div>
        Chewam Â© 2020
        <Spacer />
        <Trans id="INSEE data" /> (
        <a
          target="_blank"
          rel="noopener noreferrer"
          href="https://www.data.gouv.fr/fr/datasets/fichier-des-personnes-decedees"
        >
          data.gouv.fr
        </a>
        )
      </div>

      <div className="wrapper">
        <Switch />
        <Spacer />
        <Lang />
        <Spacer />
        version {version} (
        <a
          target="_blank"
          rel="noopener noreferrer"
          href={`https://github.com/chewam/deaths/tree/${sha}`}
        >
          {sha.substring(0, 7)}
        </a>
        )
      </div>
    </footer>
  )
}

export default Footer
