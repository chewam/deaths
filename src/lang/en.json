{
  "overview": "Overview",
  "comparison": "Comparison",
  "distribution": "Distribution",
  "Mortality rate": "Mortality rate",
  "Deaths count": "Deaths count",
  "Total population": "Total population",
  "Mortality in France": "Mortality in France",
  "INSEE data": "INSEE data",
  "deaths": "deaths",
  "males": "males",
  "females": "females",
  "Age group": "Age group",
  "to": "to",
  "years old": "years old",
  "light mode": "light mode",
  "dark mode": "dark mode",
  "January": "January",
  "February": "February",
  "March": "March",
  "April": "April",
  "May": "May",
  "June": "June",
  "July": "July",
  "August": "August",
  "September": "September",
  "October": "October",
  "November": "November",
  "December": "Decembere"
}
