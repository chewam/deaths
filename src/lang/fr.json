{
  "overview": "Vue d'Ensemble",
  "comparison": "Comparaison",
  "distribution": "Répartition",
  "Mortality rate": "Taux de mortalité",
  "Deaths count": "Nombre de décès",
  "Total population": "Population totale",
  "Mortality in France": "Mortalité en France",
  "INSEE data": "Données INSEE",
  "deaths": "décès",
  "males": "hommes",
  "females": "femmes",
  "Age group": "Tranche d'âge",
  "to": "à",
  "years old": "ans",
  "light mode": "mode clair",
  "dark mode": "mode sombre",
  "January": "Janvier",
  "February": "Février",
  "March": "Mars",
  "April": "Avril",
  "May": "Mai",
  "June": "Juin",
  "July": "Juillet",
  "August": "Aout",
  "September": "Septembre",
  "October": "Octobre",
  "November": "Novembre",
  "December": "Decembre"
}
