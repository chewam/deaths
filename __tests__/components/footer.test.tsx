import fr from "@/lang/fr.json"
import en from "@/lang/en.json"
import Footer from "@/components/Footer"
import { IntlProvider } from "react-intl"
import { render } from "@testing-library/react"

const messages = { en, fr }

jest.mock("next/router", () => ({
  useRouter() {
    return {
      locale: "fr",
      pathname: "/",
      defaultLocale: "en",
    }
  },
}))

/* eslint @typescript-eslint/no-var-requires: "off" */
const useRouter = jest.spyOn(require("next/router"), "useRouter")

describe("Footer", () => {
  test("should create a Footer in french", () => {
    const lang = "fr"

    useRouter.mockImplementation(() => ({
      locale: lang,
      pathname: "/",
    }))

    const { asFragment } = render(
      <IntlProvider locale={lang} messages={messages[lang]}>
        <Footer />
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should create a Footer in english", () => {
    const lang = "en"

    useRouter.mockImplementation(() => ({
      locale: lang,
      pathname: "/",
    }))

    const { asFragment } = render(
      <IntlProvider locale={lang} messages={messages[lang]}>
        <Footer />
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
