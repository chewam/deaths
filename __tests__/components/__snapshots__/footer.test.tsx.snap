// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should create a Footer in english 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="left"
    >
      Chewam © 2020
      <span
        class="spacer"
      >
          |  
      </span>
      INSEE data (
      <a
        href="https://www.data.gouv.fr/fr/datasets/fichier-des-personnes-decedees"
        rel="noopener noreferrer"
        target="_blank"
      >
        data.gouv.fr
      </a>
      )
    </div>
    <div
      class="right"
    >
      <div
        class="switch"
      >
        <i
          class="ri-moon-line"
        />
      </div>
      <span
        class="spacer"
      >
          |  
      </span>
      <a
        class="lang"
        href="/"
      >
        fr
      </a>
      <span
        class="spacer"
      >
          |  
      </span>
      version 0.0.0 (
      <a
        href="https://github.com/chewam/deaths/tree/dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        dev
      </a>
      )
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Footer should create a Footer in french 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="left"
    >
      Chewam © 2020
      <span
        class="spacer"
      >
          |  
      </span>
      Données INSEE (
      <a
        href="https://www.data.gouv.fr/fr/datasets/fichier-des-personnes-decedees"
        rel="noopener noreferrer"
        target="_blank"
      >
        data.gouv.fr
      </a>
      )
    </div>
    <div
      class="right"
    >
      <div
        class="switch"
      >
        <i
          class="ri-moon-line"
        />
      </div>
      <span
        class="spacer"
      >
          |  
      </span>
      <a
        class="lang"
        href="/"
      >
        en
      </a>
      <span
        class="spacer"
      >
          |  
      </span>
      version 0.0.0 (
      <a
        href="https://github.com/chewam/deaths/tree/dev"
        rel="noopener noreferrer"
        target="_blank"
      >
        dev
      </a>
      )
    </div>
  </footer>
</DocumentFragment>
`;
